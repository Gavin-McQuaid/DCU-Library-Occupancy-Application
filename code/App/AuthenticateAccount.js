import React, {Component} from 'react';
import {View,Text, Alert, TextInput, StyleSheet, Image,KeyboardAvoidingView, TouchableOpacity} from 'react-native';
import {Ionicons} from '@expo/vector-icons';
//Comment line above and uncomment line below if you want to run integration tests using Cavy and the iOS simulator
//import Ionicons from 'react-native-vector-icons/Ionicons';
import {hook} from 'cavy';
import { NavigationActions } from 'react-navigation';
class AuthenticateAccount extends Component {
	static navigationOptions = {
		title : 'AuthenticateAccount'
	}
	constructor(props){
		super(props)
		this.state = {
			AuthenticationCode: ''
		}
	}

    // Function that authenticates a code with the database when the code is entered into the text field and submitted
	AuthenticateCode = () => {
		const {AuthenticationCode} = this.state;
		const resetAction = NavigationActions.reset({
  index: 0,
  actions: [NavigationActions.navigate({ routeName: 'Menu', params: {email: this.props.navigation.state.params.email}})],
});
		fetch('http://ec2-34-244-240-250.eu-west-1.compute.amazonaws.com/AuthenticateCode.php', {
			method: 'POST',
			headers: {
				'Accept' : 'application/json',
				'Content-Type' : 'applcation/json'
			},
			body: JSON.stringify({
				code: AuthenticationCode,
				email: this.props.navigation.state.params.email
			}) 
		}).then((response) => response.json())
      .then((responseJson) => {
		if(responseJson == "Success"){
			this.props.navigation.dispatch(resetAction);
		}
		else{
			Alert.alert("That code is incorrect or expired");
		}
      }).catch((error) => {
        console.error(error);
      });
	}

	// Function that triggers the generation of a new code for a given email address
	GenerateCode = () => {
		fetch('http://ec2-34-244-240-250.eu-west-1.compute.amazonaws.com/GenerateCode.php', {
			method: 'POST',
			headers: {
				'Accept' : 'application/json',
				'Content-Type' : 'applcation/json'
			},
			body: JSON.stringify({
				email: this.props.navigation.state.params.email
			}) 
		}).then((response) => response.json())
      .then((responseJson) => {
		if(responseJson == "Success"){
			Alert.alert("Check your email");
		}
		else{
			Alert.alert("Something went wrong");
		}
      }).catch((error) => {
        console.error(error);
      });}

	render() {
		return (
		<KeyboardAvoidingView behavior = 'padding' style = {styles.container}>
			<View style = {styles.container}>
				<View style = {styles.topBar}
					ref = {this.props.generateTestHook('AuthenticateAccount.topBar')}> 
					<Ionicons name="md-settings" 
						size={40} 
						color="#FFFFFF"
			    		onPress={() => this.props.navigation.navigate("Settings", 
			    			{screen: "Settings", email: this.props.navigation.state.params.email })}
			    	/>
				</View>
				<View style = {styles.logoContainer}>
					<Text style = {styles.title}> DCU Library App 
					</Text>
					<Image
						style = {styles.logo}
						source = {require('./images/logo.png')} 
					/>					
				</View>
				<TextInput
					ref = {this.props.generateTestHook('AuthenticateAccount.code')}
					placeholder = "Authentication code"
					placeholderTextColor = 'white'
					returnKeyType = 'next'
					autoCapitalize = 'none'
					autoCorrect = {false}
					onChangeText ={AuthenticationCode => this.setState({AuthenticationCode})}
					style = {styles.input}
				/>		
				<TouchableOpacity
					ref = {this.props.generateTestHook('AuthenticateAccount.codeButton')}
					style = {styles.buttonContainer}
					onPress={() => this.AuthenticateCode()}>
					<Text style = {styles.buttontext}> Confirm authentication code 
					</Text>
				</TouchableOpacity>
				<TouchableOpacity	
					style = {styles.buttonContainer}
					onPress={() => this.GenerateCode()}
					ref = {this.props.generateTestHook('AuthenticateAccount.newcodeButton')}>
					<Text style = {styles.buttontext}> Generate new code 
					</Text>
				</TouchableOpacity>
			</View>
		</KeyboardAvoidingView>	
		);
	}
}

export default hook(AuthenticateAccount);
const styles = StyleSheet.create({
  container: {
	backgroundColor : 'cornflowerblue',  
    flex: 1,
  },

  logoContainer: {
	alignItems: 'center',
	flexGrow: 1,
	justifyContent : 'center'
  },

  logo: {
	width : 200,
	height : 150
  },

  title: {
	color : 'white',
	marginBottom : 10,
	fontSize : 30,
	opacity : 0.9,
	justifyContent: 'center'	
  },

  input: { 
	height: 40,
	backgroundColor : 'rgba(255,255,255,0.2)',
	marginBottom : 15,
	color : 'white',
	paddingHorizontal : 10,
	borderRadius : 25
	},

   buttonContainer : {
	backgroundColor :'steelblue',
	paddingVertical : 15,
	marginBottom : 15,
	borderRadius : 25
	},

  buttontext : {
	color : 'white',
	textAlign : 'center',
	fontWeight : '400'
	},

  topBar: {
	backgroundColor : 'cornflowerblue',  
	height:50,
	alignSelf: 'flex-end',
	paddingRight: 3,
	}
});