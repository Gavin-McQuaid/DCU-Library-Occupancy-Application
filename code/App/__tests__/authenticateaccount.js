import React from 'react';
import AuthenticateAccount from '../AuthenticateAccount';
import renderer from 'react-test-renderer';
import {shallow, mount,render,configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import mockStore from 'redux-mock-store';
const Text = require('Text');
const TouchableHighlight = require('TouchableHighlight');
const ReactTestRenderer = require('react-test-renderer');
const TouchableOpacity = require('TouchableOpacity');
const TextInput = require('TextInput');
const Image = require('Image');

var ReactTestUtils = require('react-dom/test-utils')
configure({ adapter: new Adapter() })

describe('AuthenticateAccount', () =>{
	it('text shows', () =>{
		const instance0 = ReactTestRenderer.create(
			<Text style = {{}}> DCU Library App </Text>	
		);
		});
	it('image shows', () =>{
		const instance1 = ReactTestRenderer.create(
			<Image style = {{}} source = {require('./Images/logo.png')}> </Image>
		);
		
	});
	it('TextInput 1', () => {
		const instance2 = ReactTestRenderer.create(
		<TextInput placeholder = 'Authentication code'></TextInput>
		);	
	});
	it('touchable opacity shows', () =>{
		const instance3 = ReactTestRenderer.create(
			<TouchableOpacity style ={{}}>
			<Text> Confirm authentication code </Text>
			</TouchableOpacity>
		);
	});
	it('touchable opacity shows', () =>{
		const instance4 = ReactTestRenderer.create(
			<TouchableOpacity style ={{}}>
			<Text>  Generate new code </Text>
			</TouchableOpacity>
		);
	});
	it ('Text Inputs works', () => {
		const wrapper = shallow(<AuthenticateAccount />);
		const TextInput1= wrapper.find('TextInput').at(0);
		const credentials = { password : '000000'}
		TextInput1.value = credentials.password;
		expect(TextInput1.value).toBe('000000');
	});
	it ('Check AuthenticateAccount works', () => {
		const wrapper = shallow(<AuthenticateAccount />);
		const TextInput1= wrapper.find('TextInput').at(0);
		const credentials = { password : '000000'}
		TextInput1.value = credentials.password;
		expect(TextInput1.value).toBe('000000');
		const touchOpacity1 = wrapper.find('TouchableOpacity').at(0);
		const store = mockStore({});
		const navigation = {navigate : jest.fn()};
		expect(renderer.create(<AuthenticateAccount store = {store} navigation = {navigation} />)).toMatchSnapshot();
	});
	const rendered = renderer.create(<AuthenticateAccount/>).toJSON();
		expect(rendered).toMatchSnapshot();
});	
