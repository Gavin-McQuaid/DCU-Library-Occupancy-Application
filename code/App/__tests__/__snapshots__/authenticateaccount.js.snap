// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<View
  keyboardVerticalOffset={0}
  onLayout={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "cornflowerblue",
        "flex": 1,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "cornflowerblue",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "flex-end",
          "backgroundColor": "cornflowerblue",
          "height": 50,
          "paddingRight": 3,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexGrow": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 30,
            "justifyContent": "center",
            "marginBottom": 10,
            "opacity": 0.9,
          }
        }
      >
         DCU Library App
      </Text>
      <Image
        source={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "height": 150,
            "width": 200,
          }
        }
      />
    </View>
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      autoCorrect={false}
      onChangeText={[Function]}
      placeholder="Authentication code"
      placeholderTextColor="white"
      returnKeyType="next"
      style={
        Object {
          "backgroundColor": "rgba(255,255,255,0.2)",
          "borderRadius": 25,
          "color": "white",
          "height": 40,
          "marginBottom": 15,
          "paddingHorizontal": 10,
        }
      }
    />
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "steelblue",
          "borderRadius": 25,
          "marginBottom": 15,
          "opacity": 1,
          "paddingVertical": 15,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontWeight": "400",
            "textAlign": "center",
          }
        }
      >
         Confirm authentication code
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "steelblue",
          "borderRadius": 25,
          "marginBottom": 15,
          "opacity": 1,
          "paddingVertical": 15,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontWeight": "400",
            "textAlign": "center",
          }
        }
      >
         Generate new code
      </Text>
    </View>
  </View>
</View>
`;

exports[`AuthenticateAccount Check AuthenticateAccount works 1`] = `
<View
  keyboardVerticalOffset={0}
  onLayout={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "cornflowerblue",
        "flex": 1,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "cornflowerblue",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "flex-end",
          "backgroundColor": "cornflowerblue",
          "height": 50,
          "paddingRight": 3,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexGrow": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 30,
            "justifyContent": "center",
            "marginBottom": 10,
            "opacity": 0.9,
          }
        }
      >
         DCU Library App
      </Text>
      <Image
        source={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "height": 150,
            "width": 200,
          }
        }
      />
    </View>
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      autoCorrect={false}
      onChangeText={[Function]}
      placeholder="Authentication code"
      placeholderTextColor="white"
      returnKeyType="next"
      style={
        Object {
          "backgroundColor": "rgba(255,255,255,0.2)",
          "borderRadius": 25,
          "color": "white",
          "height": 40,
          "marginBottom": 15,
          "paddingHorizontal": 10,
        }
      }
    />
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "steelblue",
          "borderRadius": 25,
          "marginBottom": 15,
          "opacity": 1,
          "paddingVertical": 15,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontWeight": "400",
            "textAlign": "center",
          }
        }
      >
         Confirm authentication code
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "steelblue",
          "borderRadius": 25,
          "marginBottom": 15,
          "opacity": 1,
          "paddingVertical": 15,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontWeight": "400",
            "textAlign": "center",
          }
        }
      >
         Generate new code
      </Text>
    </View>
  </View>
</View>
`;
